### PR TITLE
Add ability to specify scheduled delivery time when sending push notifications.

### PR DESCRIPTION
**Pull request by nlutsenko**
Matching functionality for https://github.com/ParsePlatform/Parse-SDK-dotNET/pull/94
Also added tests to validate all the new logic.
cc @stanleyw @grantland

### CODE REVIEW

File: Parse/Internal/Push/State/PFPushState.m [hunk 1]
```diff
@@ -45,6 +45,19 @@ + (instancetype)stateWithState:(PFPushState *)state {
 #pragma mark - NSCopying
 ///--------------------------------------
 
+- (void)setPushDate:(NSDate *)pushDate {
+    if (self.pushDate != pushDate) {
+        NSTimeInterval interval = [pushDate timeIntervalSinceNow];
+        PFParameterAssert(interval > 0, @"Can't set the scheduled push time in the past.");
+        PFParameterAssert(interval <= 60 * 24 * 14, @"Can't set the schedule push time more than two weeks from now.");
```
Comment (richardjrossiii) [inline Parse/Internal/Push/State/PFPushState.m]: Change assertion message to 'fortnight from now'? (always wanted to have an excuse to use 'fortnight' :D)

File: Tests/Unit/PushCommandTests.m [hunk 2]
```diff
@@ -23,7 +23,7 @@ @implementation PushCommandTests
 
 - (void)testEmptyPushCommand {
     PFMutablePushState *state = [[PFMutablePushState alloc] init];
-
+
```
Comment (richardjrossiii) [inline Tests/Unit/PushCommandTests.m]: Clean this entire file up, please.

File: Parse/Internal/Push/State/PFPushState.m (hunk 1 above)
Comment (nlutsenko) [inline Parse/Internal/Push/State/PFPushState.m]: Haha. Sure why not.

File: Tests/Unit/PushStateTests.m [hunk 3]
```diff
@@ -50,29 +53,29 @@ - (void)testInit {
     XCTAssertNil(state.queryState);
     XCTAssertNil(state.expirationDate);
     XCTAssertNil(state.expirationTimeInterval);
-    XCTAssertNil(state.expirationTimeInterval);
-
+    XCTAssertNil(state.pushDate);
+
```
Comment (richardjrossiii) [inline Tests/Unit/PushStateTests.m]: And this file too.

File: Tests/Unit/PushStateTests.m [hunk 4]
```diff
@@ -120,9 +123,15 @@ - (void)testSetPayloadWithMessage {
...
     XCTAssertEqualObjects(state.payload, @{ @"alert" : @"yolo" });
-
+    
     [state setPayloadWithMessage:nil];
     XCTAssertNil(state.payload);
 }
 
+- (void)testSetPushTimeValidation {
+    PFMutablePushState *state = [[PFMutablePushState alloc] init];
+    PFAssertThrowsInvalidArgumentException(state.pushDate = [NSDate dateWithTimeIntervalSinceNow:-10]);
```
Comment (richardjrossiii) [inline Tests/Unit/PushStateTests.m]: These are the kind of tests that could end up being flaky on travis if we get a stupidly slow VM. 

I'd strongly prefer using a concrete date in the past (maybe the unix epoch) for the first test, and `distantFuture` for the second test.

**Pull request merged**